modules/standard/Sets.chpl

```diff
@@ -155,15 +166,19 @@ module Sets {
     pragma "no doc"
     inline proc _enter() {
```

Comment:
The entire operation should be contained within an `on this` statement, just for added clarity. If you have to perform two active messages just to acquire and release per operation, you're better off just making the whole operation a single active message, especially since all other accesses will be PUTs and GETs by default.

Reply:
My rationale here was that since `parSafe` is param, the entire method would be marked for removal / a NOP on the `false` path (an empty body would mark the method for removal). Though, maybe my assumption is wrong.

Reply:
It will, but I'm saying any operation that looks like...

```chpl
_enter(); // Round-Trip
// does stuff
_leave(); // Round-Trip
```

Can be better written as

```chpl
on this {
   _enter();
   // Does stuff
   _leave();
} // Round-trip
```

My comment has more to do with using `on this` solely to acquire a lock, when in reality it _should_ be for the entire operation.